assets/js/script.js

```diff
@@ -4,124 +4,80 @@ $(document).ready(function() {
   }).then(function(html) {
     html = html.replace(/assets\/img\//g, (window.baseUrl || '') + 'assets/img/')
     $('#custombar').before(html).remove()
-    activate()
+    render()
   })
 })
 
-function activate() {
-  var el = {
-    input: $('#letters-input'),
-    letterTabs: $('.tab-letter-color-letter'),
-    letterTab1: $('#tab-letter1'),
-    letterTab2: $('#tab-letter2'),
-    lettersWrapper: $('.tab-letter-wrapper.active'),
-    swatch: $('.swatch-image-border'),
-  }
-
-  Object.keys(el).forEach(function(key) {
-    if (!el[key].length) { console.warn(`No element present for ${key}`); }
-  })
-
-  var bagContainerHeight = $(".main-bag-container").width() * .7
-  $(".main-bag-container").height(bagContainerHeight)
-  var bagContainerHeight = $(".main-letters-container").width() * .7
-  $(".main-letters-container").height(bagContainerHeight)
-  var letterTabHeight = $(".swatch-row").height() / 2
-  $(".custom-letter-container").height(letterTabHeight)
+var state = {
+  letters:      '',
+  materials:    ['white', 'white'],
+  activeLetter: 0,
+}
 
-  $(window).resize(function() {
-    var newLetterHeight = $(".letters-wrapper ").height()
-    $("#bagLetter1").css({ display: "block", height: newLetterHeight + "px" })
-    $("#bagLetter2").css({ display: "block", height: newLetterHeight + "px" })
-    bagContainerHeight = $(".main-bag-container").width() * .7
-    $(".main-bag-container").height(bagContainerHeight)
-    bagContainerHeight = $(".main-letters-container").width() * .7
-    $(".main-letters-container").height(bagContainerHeight)
-    letterTabHeight = $(".swatch-row").height() / 2
-    $(".custom-letter-container").height(letterTabHeight)
+$(document).on('input', '.js-input', function(evt) {
```

Comment:
you can see that each event handler now is a minimal wrapper calling `setState`. `setState` will then deal with re-rendering the DOM according to the updated state, so we don't have to mix event handling and DOM manipulation.